store/src/java/com/zimbra/cs/service/mail/SendShareNotification.java

```diff
@@ -745,5 +765,83 @@ public void run() {
         senderThread.start();
     }
 
-}
+    private void processDBFilesSharedWithMe(MailItem reqItem, Action action, OperationContext octxt, String granteeId,
+            String fileName, String ownerAccountId, int ownerFileId, String fileUUID, String fileOwnerName,
+            String rights, Map<String, Object> context) throws ServiceException {
+        // Mailbox mbox = MailboxManager.getInstance().getMailboxByAccountId(granteeId);
+        // Account target = Provisioning.getInstance().get(Key.AccountBy.id,
+        // mRemoteOwnerId);
+        Account acct = Provisioning.getInstance().getAccountById(granteeId);
+        if (Provisioning.onLocalServer(acct)) {
+            Mailbox mbox = MailboxManager.getInstance().getMailboxByAccountId(granteeId);
+            // if file is shared, this file should be available at receivers end
+            if (reqItem.getType().equals(Type.DOCUMENT) && action == null) {
+                Document docReqItem = (Document) reqItem;
+                mbox.createFileSharedWithMe(mbox, octxt, FolderConstants.ID_FOLDER_FILE_SHARED_WITH_ME, fileName,
+                        ownerAccountId, ownerFileId, fileUUID, fileOwnerName, docReqItem.getContentType(),
+                        docReqItem.getSize(), rights);
+            } // if file access is revoked at source delete it from receiver end as well
+            else if (reqItem.getType().equals(Type.DOCUMENT) && action != null && action.name().equals(REVOKE)) {
+                String fileUUIDToDelete = fileUUID + ":" + ownerFileId;
+                mbox.deleteFileSharedWithMe(mbox, octxt, fileUUIDToDelete);
+            } // if file rights is changed , update file permission accordingly
+            else if (reqItem.getType().equals(Type.DOCUMENT) && action != null
+                    && action.name().equalsIgnoreCase(EDIT)) {
+                Document docReqItem = (Document) reqItem;
+                mbox.updateFileSharedWithMe(mbox, octxt, ownerAccountId, ownerFileId, fileUUID, fileOwnerName,
+                        docReqItem.getContentType(), rights);
+            }
+        } else {
+            Document docReqItem = (Document) reqItem;
```

Comment:
Cant't we use `proxyRequest`?

Reply:
here the request is getting changed from SendShareNotification to FileSharedWithMe
@gabhinavTI did you try proxyRequest here instead since the new request can be formulated here to pass on along with the current context from the handle of SendShareNotification?

Reply:
tried using proxyRequest here, but I am coming across errors . hence proceeding with sendRequest call which works for both single node and multi node environment without any issue

Reply:
used proxyRequest , some changes in `ZimbraSoapContext` class was required . Implemented with support of Stephan